### PR TITLE
Accept non-existent session token in auth.Deauthenticate.

### PR DESCRIPTION
**Pull request by aapeliv**
Also sends an empty cookie expiring now() to remove it from the browser as an extra.

### CODE REVIEW

**Comment by aapeliv** [comment]
I've also cleaned the tests a little bit so they give nicer error messages... just because...

File: app/backend/src/tests/test_auth.py [hunk 1]
```diff
@@ -204,4 +204,35 @@ def test_password_reset_invalid_token(db, fast_passwords):
...
+
+    reply_token = get_session_cookie_token(metadata_interceptor)
+
+    # delete all login tokens
+    with session_scope() as session:
+        session.query(LoginToken).delete()
+
+    # log out with non-existent token should still return a valid result
+    with auth_api_session() as (auth_api, metadata_interceptor):
+        reply = auth_api.Deauthenticate(empty_pb2.Empty(), metadata=(("cookie", f"couchers-sesh={reply_token}"),))
```
Comment (kalvdans) [inline app/backend/src/tests/test_auth.py]: unused variable; test `reply` or remove the assigment

File: app/backend/src/couchers/servicers/auth.py [hunk 2]
```diff
@@ -354,11 +354,18 @@ def Deauthenticate(self, request, context):
...
-            # probably caused by token not existing
-            context.abort(grpc.StatusCode.UNKNOWN, errors.LOGOUT_FAILED)
+        # if we had a token, try to remove the session
+        if token:
+            self._delete_session(token)
+
+        # set the cookie to an empty string and expire immediately, should remove it from the browser
+        context.send_initial_metadata(
+            [
+                ("set-cookie", create_session_cookie("", now())),
```
Comment (kalvdans) [inline app/backend/src/couchers/servicers/auth.py:364]: Would be easier if we used Max-Age insead of Expires, but that's a separate change.

**Review by kalvdans — APPROVED** [review]
LGTM!
I don't think there was any issue to close.

**Pull request merged**